src/WebJobs.Script/Binding/GeneralScriptBindingProvider.cs

```diff
@@ -72,7 +72,7 @@ internal static Type GetRequestedType(ScriptBindingContext context)
             {
                 // arrays are supported for both trigger input as well
                 // as output bindings
-                type = type.GetType() == typeof(ParameterBindingData) ? typeof(IEnumerable<>).MakeGenericType(type) : type.MakeArrayType();
+                type = context.SupportsDeferredBinding ? typeof(IEnumerable<ParameterBindingData>) : type.MakeArrayType();
```

Comment:
Need to be careful with this - supportsDeferredBinding could be set to true, but then skipDeferredBinding might be added in the expression case so `supportsDeferredBinding` might be misleading - May need to check both flags. Theres also an extension util for this

Reply:
I agree. We will also need similar change on line 63 above, right?

```csharp
Type type = context.SupportsDeferredBinding ? typeof(ParameterBindingData) : ParseDataType(context);
```

Reply:
no change is needed in this class for collection support. Let's discuss offline about correcting line 63. Will create new PR if needed.